app/models/agents/slack_agent.rb

```diff
@@ -67,7 +71,11 @@ def slack_notifier
     def receive(incoming_events)
       incoming_events.each do |event|
         opts = interpolated(event)
-        slack_notifier.ping opts[:message], channel: opts[:channel], username: opts[:username]
+        if /^:/.match(opts[:icon])
+            slack_notifier.ping opts[:message], channel: opts[:channel], username: opts[:username], icon_emoji: opts[:icon]
+        else
```

Comment:
Do you want an `elsif` when it's not `present?` to handle the default case?

Also, this is minor, but please indent 2 spaces, not 4.